internal/fs/buffer/inmemory_write_buffer_test.go

```diff
@@ -41,24 +46,240 @@ func (t *MemoryBufferTest) SetUp(ti *TestInfo) {
 
 func (t *MemoryBufferTest) TearDown() {}
 
+// //////////////////////////////////////////////////////////////////////
+// Helpers
+// //////////////////////////////////////////////////////////////////////
+
+func generateRandomData(size int64) []byte {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	data := make([]byte, size)
+	r.Read(data)
```

Comment:
Pass `t *MemoryBufferTest` to this function, check the output value of r.Read() and assert on that using `t`. Not doing this adds to our already big list of linter/static-check errors.